### PR TITLE
Support EIP-2930 RLP encoding

### PR DESCRIPTION
**Pull request by sheapwilliams**
[EIP-2930](https://eips.ethereum.org/EIPS/eip-2930) specification, and some [discussion](https://www.youtube.com/watch?v=qQpvkxKso2E) on it.

This is currently rebased onto #20 , as it evolves that code.

### CODE REVIEW

File: include/monad/core/signature.hpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+#include <optional>
+
+MONAD_NAMESPACE_BEGIN
+
+struct SignatureAndChain
+{
+    uint256_t r{};
+    uint256_t s{};
+    std::optional<uint64_t> chain_id{};
+    bool odd_y_parity{};
```
Comment (sheapwilliams) [inline include/monad/core/signature.hpp:14]: Chain ID and parity are gathered here in to the same structure cause [EIP-155](https://eips.ethereum.org/EIPS/eip-155) coupled them in `v`.

File: src/monad/rlp/encode_helpers.cpp [hunk 2]
```diff
@@ -33,7 +55,19 @@ byte_string encode(Transaction const &txn)
             encode_unsigned(txn.sc.r),
             encode_unsigned(txn.sc.s));
     }
-    return encode_list();
+    MONAD_ASSERT(txn.sc.chain_id != std::nullopt);
+    return byte_string{0x01} += encode_list(
```
Comment (zander-xyz) [inline src/monad/rlp/encode_helpers.cpp:59]: can we switch on the transaction type and then just `assert(false)` or something for invalid values?

Comment (sheapwilliams) [reply src/monad/rlp/encode_helpers.cpp:59]: Yea, we can do that...

Comment (sheapwilliams) [reply src/monad/rlp/encode_helpers.cpp:59]: I'll address that in #27 - the 2930 logic is pretty simple.

File: src/monad/rlp/encode_helpers.cpp [hunk 3]
```diff
@@ -1,9 +1,34 @@
...
+#include <algorithm>
+#include <numeric>
+#include <string>
 
 MONAD_RLP_NAMESPACE_BEGIN
 
+byte_string encode_access_list(Transaction::AccessList const &list)
+{
+    byte_string result{};
+    std::ranges::for_each(list, [&result](auto const &i) {
```
Comment (jhunsaker) [inline src/monad/rlp/encode_helpers.cpp:17]: `for (auto const& i : list) {` is easier to read?

Comment (sheapwilliams) [reply src/monad/rlp/encode_helpers.cpp:17]: Sure - see #29

**Pull request merged**